corehq/apps/userreports/tasks.py

```diff
@@ -138,8 +146,67 @@ def rebuild_indicators_in_place(indicator_config_id, initiated_by=None, source=N
             config.meta.build.rebuilt_asynchronously = False
             config.save()
 
-        adapter.build_table(initiated_by=initiated_by, source=source)
-        _iteratively_build_table(config, in_place=True)
+        rows_count_before_rebuild = _get_rows_count_from_existing_table(adapter) if adapter.table_exists else None
+        try:
+            adapter.build_table(initiated_by=initiated_by, source=source)
+            _iteratively_build_table(config, in_place=True)
+        except Exception:
+            _report_ucr_rebuild_metrics(config, source, 'rebuild_datasource_in_place', adapter,
+                                        rows_count_before_rebuild, error=True)
+            raise
+        _report_ucr_rebuild_metrics(config, source, 'rebuild_datasource_in_place', adapter,
+                                    rows_count_before_rebuild)
+
+
+def _get_rows_count_from_existing_table(adapter):
+    table = adapter.get_existing_table_from_db()
+    return adapter.session_helper.Session.query(table).count()
+
+
+def _report_ucr_rebuild_metrics(config, source, action, adapter, rows_count_before_rebuild, error=False):
+    if source not in ('edit_data_source_rebuild', 'edit_data_source_build_in_place'):
+        return
+    try:
+        _report_metric_number_of_days_since_first_build(config.domain, config.get_id, action)
+        if error:
+            _report_metric_rebuild_error(config, action)
+        if not error:
+            _report_metric_increase_in_rows_count(config, action, adapter, rows_count_before_rebuild)
```

Comment:
Can be refactored:
> if error:
...
else:
...

It just reads better.

Reply:
Addressed in f4bc32e5a3ce4627271a28f6f2f2b83c57621ada